src/Parsing/Impl/Definitions/IBindableNode.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.Python.Parsing.Ast;
+
+namespace Microsoft.Python.Parsing {
+    internal interface IBindableNode : IScopeNode {
+        #region Binding
+        void Bind(PythonNameBinder binder);
+
+        void FinishBind(PythonNameBinder binder);
+
+        bool TryBindOuter(IBindableNode from, string name, bool allowGlobals, out PythonVariable variable);
+        #endregion
+
+        #region Add variables
+        void AddVariable(PythonVariable variable);
+        void AddFreeVariable(PythonVariable variable, bool accessedInScope);
+
+        string AddReferencedGlobal(string name);
+
+        void AddNonLocalVariable(NameExpression name);
+
+        void AddCellVariable(PythonVariable variable);
+        #endregion
+
+        #region References
+        PythonVariable BindReference(PythonNameBinder binder, string name);
+
+        PythonReference Reference(string name);
+
+        bool IsReferenced(string name);
+        #endregion
+
+        #region Create Variables
+        PythonVariable CreateVariable(string name, VariableKind kind);
+
+        PythonVariable EnsureVariable(string name);
+
+        PythonVariable EnsureGlobalVariable(string name);
+
+        PythonVariable DefineParameter(string name);
+        #endregion
+
+        #region Contains
+        bool ContainsImportStar { get; set; }
+        bool ContainsExceptionHandling { get; set; }
+
+        bool ContainsUnqualifiedExec { get; set; }
```

Comment:
Can I expose these in `IScopeNode`? `IScopeNode` already has a few members similar to these and it would simplify the code in `PythonNameBinder` because I wouldn't have to cast.

Reply:
What casts are you referring to? I don't see any more around these specific properties than existed previously in the binder.

Reply:
I would prefer to about it, as it's more of an implementation detail of the parser/binder rather than a property of the scope.

Reply:
I created a new property `BoundScope` which casts the scope to an `IBindableNode`. But I'll leave it in there if you feel it's best.